pcdsdevices/smarpod.py

```diff
@@ -0,0 +1,266 @@
+    cmd_rz = Cpt(EpicsSignal, ':CMD:RZ', kind='normal')
+
+
+
+    """
+    Stop
+    """
+    cmd_stop = Cpt(EpicsSignal, ':CMD:STOP', kind='normal')
+
+
+
+    """
+    Pose reachable?
+    """
+    cmd_reachable = Cpt(EpicsSignal, ':CMD:REACHABLE', kind='normal')
+    reachable = Cpt(EpicsSignal, ':REACHABLE', kind='normal')
+
+
+
+    """
+    Positions - Readback
+    """
+    cmd_pos_rbv = Cpt(EpicsSignal, ':CMD:POS_RBV', kind='normal', doc='Readback current pose position')
+    x_m = Cpt(EpicsSignal, ':X_M', kind='normal')
+    y_m = Cpt(EpicsSignal, ':Y_M', kind='normal')
+    z_m = Cpt(EpicsSignal, ':Z_M', kind='normal')
+    x = Cpt(EpicsSignal, ':X', kind='normal')
+    y = Cpt(EpicsSignal, ':Y', kind='normal')
+    z = Cpt(EpicsSignal, ':Z', kind='normal')
+    rx = Cpt(EpicsSignal, ':RX', kind='normal')
+    ry = Cpt(EpicsSignal, ':RY', kind='normal')
+    rz = Cpt(EpicsSignal, ':RZ', kind='normal')
+
+
+
+    """
+    Movement - Readback
+    """
+    moving = Cpt(EpicsSignal, ':MOVING', kind='normal', doc='Movement status')
+
+
+
+    """
+    Movement - Sync
+    """
+    cmd_sync = Cpt(EpicsSignal, ':CMD:SYNC', kind='normal')
+    error_code = Cpt(EpicsSignal, ':ERROR_CODE', kind='normal')
+    error_desc = Cpt(EpicsSignal, ':ERROR_DESC', kind='normal')
+
+
+
+    """
+    Pivot-point
+    """
+    cmd_pivot_rbv = Cpt(EpicsSignal, ':CMD:PIVOT_RBV', kind='normal', doc='Readback pivot point')
+    cmd_set_pivot = Cpt(EpicsSignal, ':CMD:SET_PIVOT', kind='normal')
+    cmd_sync_pivot = Cpt(EpicsSignal, ':CMD:SYNC_PIVOT', kind='normal')
+    cmd_pivot_mode = Cpt(EpicsSignal, ':CMD:PIVOT_MODE', kind='normal')
+    pivot_mode = Cpt(EpicsSignal, ':PIVOT_MODE', kind='normal', doc='Pivot mode')
+
+
+
+    """
+    Pivot positions
+    """
+    px_m = Cpt(EpicsSignal, ':PX_M', kind='normal')
+    py_m = Cpt(EpicsSignal, ':PY_M', kind='normal')
+    pz_m = Cpt(EpicsSignal, ':PZ_M', kind='normal')
+    px = Cpt(EpicsSignal, ':PX', kind='normal')
+    py = Cpt(EpicsSignal, ':PY', kind='normal')
+    pz = Cpt(EpicsSignal, ':PZ', kind='normal')
+
+
+
+    """
+    Pivot request positions
+    """
+    cmd_px = Cpt(EpicsSignal, ':CMD:PX', kind='normal')
+    cmd_py = Cpt(EpicsSignal, ':CMD:PY', kind='normal')
+    cmd_pz = Cpt(EpicsSignal, ':CMD:PZ', kind='normal')
+
+
+
+    """
+    Miscellaneous
+    """
+    cmd_calibrate = Cpt(EpicsSignal, ':CMD:CALIBRATE', kind='normal')
+    cmd_read_ver = Cpt(EpicsSignal, ':CMD:READ_VER', kind='normal')
+    ver_sys = Cpt(EpicsSignal, ':VER:SYS', kind='normal')
+
+
+
+    """
+    Device Info
+    """
+    ver_sn  = Cpt(EpicsSignal, ':VER:SN', kind='normal')
+    ver_product  = Cpt(EpicsSignal, ':VER:PRODUCT', kind='normal')
+    ver_firmware = Cpt(EpicsSignal, ':VER:FIRMWARE', kind='normal')
+
+
+
+    """
+    Status
+    """
+    cmd_read_status = Cpt(EpicsSignal, ':CMD:READ_STATUS', kind='normal')
+    status_temp = Cpt(EpicsSignal, ':STATUS:TEMP', kind='normal')
```

Comment:
I don't think these should be "normal" but rather omitted in most cases.

See https://confluence.slac.stanford.edu/display/PCDS/How+to+Create+an+Ophyd+Device#HowtoCreateanOphydDevice-Choosingtheright%22kind%22

Reply:
the importance of the channels has been revised.